### PR TITLE
Complete-at-point

### PR DESCRIPTION
**Pull request by AnselmC**

### CODE REVIEW

**Comment by stuhlmueller** [comment]
Seems promising! 

Using Anthropic/sonnet, I get this though (self-explanatory, I think):

Emacs 2024-10-21 org 000641

Also would be nice if the model were aware of context that follows after the cursor.

File: gpt.el [hunk 1]
```diff
@@ -299,23 +303,23 @@ PROMPT-FILE is the temporary file containing the prompt."
      ;; the `define-derived-mode` macro expects a literal as its first argument
      ;; hence, we can not simply use the `parent-mode` variable
      ;; but need to use a backquoted list and eval it
-    `(define-derived-mode gpt-mode ,parent-mode "GPT"
-      "A mode for displaying the output of GPT commands."
-      (message "GPT mode intialized with parent: %s" ',parent-mode)
-      (setq-local word-wrap t)
-      (setq-local font-lock-extra-managed-props '(invisible))
-      (if (eq ',parent-mode 'markdown-mode)
+     `(define-derived-mode gpt-mode ,parent-mode "GPT"
```
Comment (AnselmC) [inline gpt.el:306]: I seemed to have forgotten to format this previously.

**Comment by AnselmC** [comment]
Ok great, I brought it into a mergable state. As suggested I'm now passing the rest of the buffer as context which seems to work quite nicely. I also fixed the anthropic issue.

**Comment by stuhlmueller** [comment]
[This](https://github.com/stuhlmueller/gpt.el/commit/a3161c37dad2304e5dbef5962190efdf28a4395a) seems to work better for me.

**Comment by AnselmC** [comment]
Yes looks a lot more reasonable. Should I add the commit on top of this?

**Comment by AnselmC** [comment]
Added it

**Comment by AnselmC** [comment]
the latest commit fixes a bug where the streamed response would overwrite itself if the next chunk came before the previous one was already inserted.
Additionally, I added a `gpt-transform-region` command

**Comment by ryukinix** [comment]
That would be nice, actually.

**Pull request closed without merging**